internal/wasm/jit/asm/amd64/golang_asm.go

```diff
@@ -16,7 +16,7 @@ type assemblerGoAsmImpl struct {
 }
 
 func newGolangAsmAssembler() (*assemblerGoAsmImpl, error) {
```

Comment:
can we move internal/wasm/jit/asm -> internal/jit. I don't think it relies on anything in wasm or jit

Also, can we make the golang/twitchy stuff dependent on debug flag or arm64 to try to reduce the deps?
ex.

```
//go:build debug_mode || arm64
```

Reply:
I will try this

Reply:
it will be too much work to unravel this. I'll put a TODO until arm64 is finished. Right now, we are sharing logic and having it compile on both architectures, which is really helpful when completing something as it controls drift. We can later schedule to decouple better, as the decoupling will make drift hard to manage